src/google/protobuf/compiler/main.cc

```diff
@@ -101,6 +105,29 @@ int ProtobufMain(int argc, char* argv[]) {
 }  // namespace protobuf
 }  // namespace google
 
-int main(int argc, char* argv[]) {
+#ifdef _MSC_VER
+std::string ToMultiByteUTF8String(const wchar_t* input)
+{
+    int size = WideCharToMultiByte(CP_UTF8, 0, input, wcslen(input), 0, 0, NULL, NULL);
+    std::string result(size, 0);
+    if (size) WideCharToMultiByte(CP_UTF8, 0, input, wcslen(input), &result[0], size, NULL, NULL);
+    return result;
+}
+
+int main(int argc, char* argv[]) 
+{
+  wchar_t** wargv = CommandLineToArgvW(GetCommandLineW(), &argc);
+  char** argv_mbcs = new char* [argc];
+  for (int i = 0; i < argc; i++)
+  {
+      auto multibyte_string = ToMultiByteUTF8String(wargv[i]);
+      argv_mbcs[i] = const_cast<char*>(multibyte_string.c_str());
```

Comment:
this is now a dangling pointer.  What you wanted here was

```
      std::string* multibyte_string = new auto(ToMultiByteUTF8String(wargv[i]));
      argv_mbcs[i] = const_cast<char*>(multibyte_string->c_str());
```

Reply:
Ah yes good catch. Fixed in force push 881ae8e322747955fe2d2351bb7f2cd5b3561758